### ISSUE TITLE
Premature optimization is evil, and missed obvious optimization

### ISSUE THREAD

**Issue by X-Ryl669** [rendro/vintageJS#22]
Hi,

  You're using optimizations technics that just obfuscate the code for no benefit (like the --i or i<<=1 instead of i++, i*= 2), and at the same time, you're missing obvious optimization where the code is spending most of the time:

```
     // curves
      if (!!_effect.curves) {
        _imageData[idx  ] = _effect.curves.r[ _imageData[idx  ] ];
        _imageData[idx+1] = _effect.curves.g[ _imageData[idx+1] ];
        _imageData[idx+2] = _effect.curves.b[ _imageData[idx+2] ];
      }

      // contrast
      if (!!_effect.contrast) {
        _imageData[idx  ] = contrastFactor * (_imageData[idx  ] - 128) + 128;
        _imageData[idx+1] = contrastFactor * (_imageData[idx+1] - 128) + 128;
        _imageData[idx+2] = contrastFactor * (_imageData[idx+2] - 128) + 128;
      }

      // brightness
      if (!!_effect.brightness) {
        _imageData[idx  ] += _effect.brightness;
        _imageData[idx+1] += _effect.brightness;
        _imageData[idx+2] += _effect.brightness;
      }

      // screen
      if (!!_effect.screen) {
        _imageData[idx  ] = 255 - ((255 - _imageData[idx  ]) * (255 - _effect.screen.r * _effect.screen.a) / 255);
        _imageData[idx+1] = 255 - ((255 - _imageData[idx+1]) * (255 - _effect.screen.g * _effect.screen.a) / 255);
        _imageData[idx+2] = 255 - ((255 - _imageData[idx+2]) * (255 - _effect.screen.b * _effect.screen.a) / 255);
      }
```

All of these only depends on the pixel value, not its position or its neighbors. So, help yourself and create a LUT containing all theses operations, and only apply this once, like this:

```
   // Do this once, out of the pixel loop.
   LUT_red = [] ... for gb
   for (var i = 0; i < 255; i++)
   {
      LUT_red[i] = 255-((255 - (contrastFactor * (_effect.curves.r[i] - 128) + 128) + _effect.brightness))*(255 - _effect.screen.r * _effect.screen.a) / 255);
      // Ditto for other colors
  }


  // Then, in the pixel loop, only do this:
  for (...pixel_loop) { 
  _imageData[idx] = LUT_red[_imageData[idx]];
```

BTW, you don't need to test if the effect is enabled PER pixel, just provide a identity default (like a contrast of 1.0, a brightness of 0, a curves with identity (f(x) = x) and a negative screen.

In other words, profile your code, spot where you spend time and only optimize here. There is no point in gaining a µs for a loop test that happens only a few time.
